apm-agent-api/src/main/java/co/elastic/apm/api/Span.java

```diff
@@ -281,7 +281,7 @@
      *
      * @param throwable the exception to report
      */
-    void captureException(Throwable throwable);
+    String captureException(Throwable throwable);
```

Comment:
Technically, this is a breaking change in binary compatibility. When not re-compiling the application and updating the dependency, it would result in an exception when calling the method. That is because methods are linked also based on their return type.
But still think we can leave it as-is as most likely, someone would re-compile the application again after updating the dependency to the agent API. 

A more conservative option would be to create a method with a different name, like `captureExceptionAndReturnErrorId` and leave `void captureException(Throwable throwable)` as-is.

Reply:
Good point! 
I agree it's best to leave as is (meaning - just change the method signature) and not to complicate the API for this corner case.

Reply:
added new method

Reply:
I think you got us wrong. While there is a corner case for a potential breaking change, just leave `captureException` as the method name to not clutter the API too much.